tree/ntuple/v7/test/ntuple_compat.cxx

```diff
@@ -58,3 +60,312 @@ TEST(RNTupleCompat, FeatureFlag)
       EXPECT_THAT(err.what(), testing::HasSubstr("unsupported format feature: 137"));
    }
 }
+
+namespace fwd_compat {
+
+class RHandcraftedKeyBlob : public TKey {
+public:
+   RHandcraftedKeyBlob(TFile *file, std::size_t keyLen) : TKey(file)
+   {
+      fClassName = RNTuple::Class_Name();
+      fKeylen = keyLen;
+   }
+
+   /// Register a new key for a data record of size nbytes
+   void Reserve(size_t nbytes, std::uint64_t *seekKey)
+   {
+      Create(nbytes);
+      *seekKey = fSeekKey;
+   }
```

Comment:
`TKey` is derived from `TObject` so it must use one of the `ClassDef` macros, probably `ClassDefOverride` in this case.

Reply:
NOTE: used version = 0 because the class is transient (it gives issues if the version is > 0)